iosApp/iosApp/Pages/Search/SearchOverlay.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  SearchOverlay.swift
+//  iosApp
+//
+//  Created by esimon on 9/25/24.
+//  Copyright © 2024 MBTA. All rights reserved.
+//
+
+import shared
+import SwiftUI
+
+struct SearchOverlay: View {
+    @ObservedObject var searchObserver: TextFieldObserver
+    @ObservedObject var nearbyVM: NearbyViewModel
+    @ObservedObject var searchVM: SearchViewModel
+
+    let globalRepository: IGlobalRepository
+    let searchResultsRepository: ISearchResultRepository
+
+    init(
+        searchObserver: TextFieldObserver,
+        nearbyVM: NearbyViewModel,
+        searchVM: SearchViewModel,
+        globalRepository: IGlobalRepository = RepositoryDI().global,
+        searchResultsRepository: ISearchResultRepository = RepositoryDI().searchResults
+    ) {
+        self.searchObserver = searchObserver
+        self.nearbyVM = nearbyVM
+        self.searchVM = searchVM
+        self.globalRepository = globalRepository
+        self.searchResultsRepository = searchResultsRepository
+    }
+
+    var body: some View {
+        ZStack(alignment: .top) {
+            if searchObserver.isSearching {
+                Color.fill2
+            }
+            VStack {
+                SearchField(searchObserver: searchObserver)
+                if searchObserver.isSearching {
+                    SearchView(
```

Comment:
suggestion (non-blocking): Looks like `SearchResultView` is already taken, but I think that would be an appropriate name here now that the query check has been pulled out of SearchView

Reply:
I was trying to avoid significant changes to anything in `SearchResultView.swift` because @BrandonTR and I started working on the result styling changes in parallel. I agree, either I could do this and possibly cause merge conflicts, or Brandon could handle it?

Reply:
I don't mind taking that on!